utils/common/src/main/java/brooklyn/util/javalang/Threads.java

```diff
@@ -44,7 +46,15 @@ public static boolean removeShutdownHook(Thread hook) {
             return Runtime.getRuntime().removeShutdownHook(hook);
         } catch (IllegalStateException e) {
             // probably shutdown in progress
-            log.debug("cannot remove shutdown hook "+hook+": "+e);
+            String text = Exceptions.collapseText(e);
+            if (text.contains("Shutdown in progress")) {
```

Comment:
i'd prefer a typed exception here, but the exception is thrown by java, not us, if you try to add a shutdown hook while a shutdown is in progress; this is just trying to clean up the log messages which are shown at the tail of the log on a shutdown

nothing against typed exceptions where useful but in general paths which might want typed exception handling i prefer the error-listener / callback pattern, avoiding throwing altogether -- for instance `Maybe.absentThrowing` and `ReferenceWithError`.